tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/microprofiile/template/ConfigPropertyTestTemplate.java

```diff
@@ -0,0 +1,162 @@
+				readFile("resources/configProperty/CdiBean1.jav_"), false, false);
+		TextEditor te = new TextEditor(CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION);
+		new WaitWhile(new EditorHasValidationMarkers(te));
+		te.save();
+		new WaitWhile(new EditorHasValidationMarkers(te));
+
+		convertToMavenProject(PROJECT_NAME, "war", false);
+		addDependency(PROJECT_NAME, "org.eclipse.microprofile.config", "microprofile-config-api", "2.0");
+		createApplicationPropertiesFile(PROJECT_NAME);
+	}
+
+	@Test
+	public void isConfigPropertyValidationAllowedTest() {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
+		pexplorer.open();
+		pexplorer.getProject(PROJECT_NAME)
+				.getProjectItem("src/main/java", PACKAGE_NAME, CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION).open();
+
+		ProblemsView pw = new ProblemsView();
+		pw.open();
+		int error_count = pw.getProblems(ProblemType.ERROR).size();
+		assertEquals("There are errors after including the Microprofile @ConfigProperty annotation into the project.",
+				error_count, 0);
+	}
+
+	private void convertToMavenProject(String projectName, String defaultPackaging, boolean withDependencies) {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
+		pexplorer.open();
+		new WaitUntil(new ProjectExists(projectName));
+		pexplorer.getProject(projectName).select();
+		new ContextMenuItem("Configure", "Convert to Maven Project").select();
+		new DefaultShell("Create new POM");
+		assertEquals("Project " + projectName + " packaging should be set to " + defaultPackaging, defaultPackaging,
+				new LabeledCombo(new DefaultGroup("Artifact"), "Packaging:").getText());
+		new PushButton("Finish").click();
+		try {
+			new DefaultShell("Convert to Maven Dependencies");
+			new WaitUntil(new ControlIsEnabled(new PushButton("Finish")), TimePeriod.LONG);
+			if (withDependencies) {
+				new PushButton("Finish").click();
+			} else {
+				new PushButton("Skip Dependency Conversion").click();
+			}
+			new WaitWhile(new ShellIsAvailable("Convert to Maven Dependencies"));
+		} catch (CoreLayerException ex) {
+
+		} finally {
+			new WaitWhile(new ShellIsAvailable("Create new POM"));
+			new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+		}
+	}
+
+	private void addDependency(String projectName, String groupId, String artifactId, String version) {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
+		pexplorer.open();
+		pexplorer.getProject(projectName).select();
+		new ContextMenuItem("Maven", "Add Dependency").select();
+		new WaitUntil(new ShellIsAvailable("Add Dependency"), TimePeriod.DEFAULT);
+		new LabeledText("Group Id:").setText(groupId);
+		new LabeledText("Artifact Id:").setText(artifactId);
+		new LabeledText("Version: ").setText(version);
+		new PushButton("OK").click();
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+
+	}
+
+	private void createApplicationPropertiesFile(String projectName) {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
```

Comment:
Is this necessary? Do you really need to use UI to create properties file?

Reply:
I have removed the createApplicationPropertiesFile() method, the whole method is not necessary right now.